core/include/detray/builders/cuboid_portal_generator.hpp

```diff
@@ -30,13 +30,14 @@ template <typename detector_t>
 class cuboid_portal_generator final
     : public surface_factory_interface<detector_t> {
 
-    using scalar_type = typename detector_t::scalar_type;
-    using transform3_type = typename detector_t::transform3_type;
+    using algebra_type = typename detector_t::algebra_type;
+    using scalar_type = dscalar<algebra_type>;
+    using transform3_type = dtransform3D<algebra_type>;
```

Comment:
I have to say, I'm not the biggest fan of these aliases to `algebra::` type traits. They very much give the impression of a hard-coded type. Even though they're not.

The `algebra::get_scalar_t`, `algebra::get_transform3D_t`, etc. traits take up more characters, but I personally would've gone with those. 🤔